core/corehttp/gateway_handler.go

```diff
@@ -194,6 +196,9 @@ func (i *gatewayHandler) optionsHandler(w http.ResponseWriter, r *http.Request)
 }
 
 func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, span := otel.Tracer("").Start(r.Context(), "gatewayHandler-getOrHeadHandler")
```

Comment:
no need to have the hypenated name here, the reader should be able to derive where this is coming from based on context within the trace viewer